yb-voyager/src/tgtdb/oracle.go

```diff
@@ -0,0 +1,478 @@
+	if db.conn_ == nil {
+		err = db.reconnect()
+		if err != nil {
+			return err
+		}
+	}
+	retry := true
+	for retry {
+		conn, err = db.conn_.Conn(context.Background())
+		if err != nil {
+			return fmt.Errorf("get connection from target db: %w", err)
+		}
+		retry, err = fn(conn)
+		if err != nil {
+			// On err, drop the connection.
+			conn.Close()
+		}
+	}
+
+	return err
+}
+
+func (db *TargetOracleDB) importBatch(conn *sql.Conn, batch Batch, args *ImportBatchArgs, exportDir string) (rowsAffected int64, err error) {
+	var file *os.File
+	file, err = batch.Open()
+	if err != nil {
+		return 0, fmt.Errorf("open batch file %q: %w", batch.GetFilePath(), err)
+	}
+	defer file.Close()
+
+	//setting the schema so that the table is created in the correct schema
+	db.setTargetSchema(conn)
+
+	ctx := context.Background()
+	var tx *sql.Tx
+	tx, err = conn.BeginTx(ctx, &sql.TxOptions{})
+	if err != nil {
+		return 0, fmt.Errorf("begin transaction: %w", err)
+	}
+	defer func() {
+		var err2 error
+		if err != nil {
+			err2 = tx.Rollback()
+			if err2 != nil {
+				rowsAffected = 0
+				err = fmt.Errorf("rollback transaction: %w (while processing %s)", err2, err)
+			}
+		} else {
+			err2 = tx.Commit()
+			if err2 != nil {
+				rowsAffected = 0
+				err = fmt.Errorf("commit transaction: %w", err2)
+			}
+		}
+	}()
+
+	var alreadyImported bool
+	alreadyImported, rowsAffected, err = db.isBatchAlreadyImported(tx, batch)
+	if err != nil {
+		return 0, err
+	}
+	if alreadyImported {
+		return rowsAffected, nil
+	}
+
+	// fmt.Printf("Importing table path: %s \n", batch.GetFilePath())
+	tableName := batch.GetTableName()
+
+	sqlldrControlFileContent := args.GetSqlLdrControlFile(db.tconf.Schema)
+
+	if _, err := os.Stat(fmt.Sprintf("%s/sqlldr", exportDir)); os.IsNotExist(err) {
+		os.Mkdir(fmt.Sprintf("%s/sqlldr", exportDir), 0755)
+	}
+	sqlldrControlFileName := fmt.Sprintf("%s.ctl", tableName)
+	sqlldrControlFilePath := fmt.Sprintf("%s/sqlldr/%s", exportDir, sqlldrControlFileName)
+	sqlldrControlFile, err := os.Create(sqlldrControlFilePath)
+	if err != nil {
+		return 0, fmt.Errorf("create sqlldr control file %q: %w", sqlldrControlFilePath, err)
+	}
+	defer sqlldrControlFile.Close()
+	_, err = sqlldrControlFile.WriteString(sqlldrControlFileContent)
+	if err != nil {
+		return 0, fmt.Errorf("write sqlldr control file %q: %w", sqlldrControlFilePath, err)
+	}
+
+	sqlldrLogFileName := fmt.Sprintf("%s.log", tableName)
+	sqlldrLogFilePath := fmt.Sprintf("%s/sqlldr/%s", exportDir, sqlldrLogFileName)
+	sqlldrLogFile, err := os.Create(sqlldrLogFilePath)
+	if err != nil {
+		return 0, fmt.Errorf("create sqlldr log file %q: %w", sqlldrLogFilePath, err)
+	}
+	defer sqlldrLogFile.Close()
+
+	// fmt.Println("Running sqlldr for file: ", sqlldrControlFilePath)
+	connectionString := db.getConnectionUri(db.tconf)
+	user := getValue(connectionString, "user")
+	password := getValue(connectionString, "password")
+	connectString := getValue(connectionString, "connectString")
```

Comment:
Consider refactoring the `getConnectionUri()` so that there is a separate `getConnectString()` function which you can directly call here.